### DISCUSSION TITLE
How to change color in `Linkplot`?

### DISCUSSION THREAD

**Discussion by jiangpuxuan** [stuart-lab/signac#1052] [Q&A] [answered]
<img width="864" alt="1647593387(1)" src="https://user-images.githubusercontent.com/81287604/158967412-d2fb1dbe-12ab-421e-80c6-5eed56b2f876.png">
I have the linkplot like this. 
The color related to score is range from 'gray' to 'blue', but it looks not so brilliant......I want to change it into `cols =c('gray', "blue",'purple','black')`, and how could I manage this?

**Comment by timoast** [comment, marked as answer]
There isn't a built-in way to do this, but the plot itself is a patchwork object, which is essentially a list of ggplot2 plots. You can modify the individual plots using standard ggplot2 functions, you just need to find which element in the patchwork list is the plot that you want to modify. In this example, you probably want the 4th element (will be different depending on what's included in the coverage plot). For example:


```r
library(Signac)
library(ggplot2)

p <- CoveragePlot(
  object = obj,
  region = "LYZ"
)

p[[1]][[4]] <- p[[1]][[4]] + scale_color_viridis_c()
p
```

<img width="783" alt="Screen Shot 2022-03-31 at 10 42 06 AM" src="https://user-images.githubusercontent.com/4591688/161082445-d9d34a60-2d2f-48c5-8d37-b89dadb1d48a.png">

**Reply by jiangpuxuan** [reply]
> There isn't a built-in way to do this, but the plot itself is a patchwork object, which is essentially a list of ggplot2 plots. You can modify the individual plots using standard ggplot2 functions, you just need to find which element in the patchwork list is the plot that you want to modify. In this example, you probably want the 4th element (will be different depending on what's included in the coverage plot). For example:
> 
> ```r
> library(Signac)
> library(ggplot2)
> 
> p <- CoveragePlot(
>   object = obj,
>   region = "LYZ"
> )
> 
> p[[1]][[4]] <- p[[1]][[4]] + scale_color_viridis_c()
> p
> ```
> 
> <img alt="Screen Shot 2022-03-31 at 10 42 06 AM" width="783" src="https://user-images.githubusercontent.com/4591688/161082445-d9d34a60-2d2f-48c5-8d37-b89dadb1d48a.png">

Thank you for your help! I will try it soon.
